python/ray/data/_internal/logical/interfaces.py

```diff
@@ -28,6 +28,12 @@ def input_dependencies(self) -> List["Operator"]:
         ), "Operator.__init__() was not called."
         return self._input_dependencies
 
+    def __iter__(self) -> Iterator["Operator"]:
```

Comment:
Actually after a second thought, shall we define the traversal method without overriding `__iter__`? Here we are doing post-order traversal, but we may need pre-order and in-order later. We can have a method called `get_post_order_iter()`

Reply:
done